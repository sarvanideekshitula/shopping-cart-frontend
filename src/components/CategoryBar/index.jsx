import React from 'react';
import PropTypes from 'prop-types';
import * as styles from './index.module.css';


const CategoryBar = (props) => {
  const { categories, buttonClick, testId } = props;
  const categoryList = categories.map((category) => <li className={styles.listelement}><button className={styles.listlink} type="button" onClick={() => buttonClick(category)} name={category}>{category}</button></li>);
  return (
    <div className={styles.CategoryBar} data-testId={testId}>
      <h3>Filter Categories</h3>
      <ul className={styles.list}>
        <li className={styles.listelement}>
          <a href="/products"><button className={styles.listlink} type="button">All</button></a>
        </li>
        {categoryList}
      </ul>
    </div>
  );
};

CategoryBar.propTypes = {
  buttonClick: PropTypes.func.isRequired,
  categories: PropTypes.objectOf.isRequired,
  testId: PropTypes.string.isRequired,
};
export default CategoryBar;
