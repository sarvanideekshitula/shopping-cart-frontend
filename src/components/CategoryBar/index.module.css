.CategoryBar{
    margin-top: 3rem;
    display: flex;
    flex-direction: row;
    margin-left: 10rem;
}

.list {
    display: flex;
    flex-direction: row;
    list-style-type: none;
    border-radius: 0.5rem;
  }
  
  .listelement {
    float: left;
  }
  
  .listlink {
    display:inline-block;
    color: black;
    text-align: center;
    background-color: rgb(196, 196, 196);
    padding: 10px 20px;
    font-size: 14px;
    text-transform: capitalize;
  }

  .listlink:hover {
    background-color: white;
    outline: none;
  }
  
  li a:hover:not(.active) {
    background-color: white;
  }
