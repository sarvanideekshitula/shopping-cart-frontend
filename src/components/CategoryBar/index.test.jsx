import React from 'react';
import { render } from '@testing-library/react';
import CategoryBar from './index';

describe('the Genre container', () => {
  it('should render correctly', () => {
    const click = jest.fn();
    const { asFragment } = render(<CategoryBar buttonClick={click} categories={[]} testId="test" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
