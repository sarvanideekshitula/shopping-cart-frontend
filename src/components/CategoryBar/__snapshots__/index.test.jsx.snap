// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Genre container should render correctly 1`] = `
<DocumentFragment>
  <div
    data-testid="test"
  >
    <h3>
      Filter Categories
    </h3>
    <ul>
      <li>
        <a
          href="/products"
        >
          <button
            type="button"
          >
            All
          </button>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
